### ISSUE TITLE
Failed async proposals do not return proposal_bond

### ISSUE THREAD

**Issue by vinibarbosabr** [near-daos/sputnik-dao-contract#238]
When an async proposal (e.g. `Transfer` or `FunctionCall`) fails during execution, the proposal transitions to `Failed` status but the `proposal_bond` is never returned to the proposer.

## Details

Proposals that require cross-contract execution optimistically set status to `Approved` and detach the execution:

```Rust
if proposal.status == ProposalStatus::Approved {
    self.internal_execute_proposal(&policy, &proposal, id).detach();
    ...
}
```

If execution fails, the callback handler only updates the status:

```Rust
// sputnikdao2/src/proposals.rs
pub(crate) fn internal_callback_proposal_fail(
    &mut self,
    proposal: &mut Proposal,
) -> PromiseOrValue<()> {
    proposal.status = ProposalStatus::Failed;
    PromiseOrValue::Value(())  // Bond is not returned
}
```

`internal_return_bonds()` is never called in the failure path. Additionally, `Finalize` does not provide a way to recover the bond for `Failed` proposals.

## Impact

+ Proposers lose their bond on technical failures (non-existent target at execution time, gas exhaustion, state changes between voting and execution, etc.).
+ No clear recovery path exists for bonds stuck in `Failed` proposals.
+ Creates an inconsistent state where the proposal appears finalized as `Approved` before the actual result is known.
+ Affects any DAO using `Transfer`, `FunctionCall`, or other async proposal kinds.

## Suggested Fix

Options to consider:

1. Return the bond inside `internal_callback_proposal_fail()` (recommended for technical failures).
2. Allow `Finalize` to return the bond when the proposal is in `Failed` state.
3. Introduce an intermediate "Executing" status to reduce the window between optimistic approval and final result.
